feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts

```diff
@@ -243,4 +255,44 @@ export class ConfiguratorGroupMenuComponent {
   isConflictGroupType(groupType: Configurator.GroupType): boolean {
     return this.configuratorGroupsService.isConflictGroupType(groupType);
   }
+
+  /**
+   * Returns group-status style classes dependent on completeness, conflicts, visited status and configurator type.
+   *
+   * @param {Configurator.Group} group - Current group
+   * @param {Configurator.Configuration} configuration - Configuration
+   * @return {Observable<boolean>} - true if visited and not a conflict group
+   */
+  getGroupStatusStyles(
+    group: Configurator.Group,
+    configuration: Configurator.Configuration
+  ): Observable<string> {
+    return this.isGroupVisited(group, configuration).pipe(
+      switchMap((isVisited) => {
```

Comment:
why `switchMap`, we can just use `map`, right? same in `isGroupVisited` function.

Reply:
picked

Reply:
Done